test/fuzz/test_runner.py

```diff
@@ -156,7 +156,7 @@ def main():
                 fuzz_bin,
                 '-help=1',
             ],
-            env=get_fuzz_env(target=test_list_selection[0], source_dir=config['environment']['SRCDIR']),
+            env={**os.environ, **get_fuzz_env(target=test_list_selection[0], source_dir=config['environment']['SRCDIR'])},
```

Comment:
The error message is:
```
Traceback (most recent call last):
  File "D:\a\bitcoin\bitcoin\test\fuzz\test_runner.py", line 406, in <module>
    main()
  File "D:\a\bitcoin\bitcoin\test\fuzz\test_runner.py", line 110, in main
    test_list_all = parse_test_list(
                    ^^^^^^^^^^^^^^^^
  File "D:\a\bitcoin\bitcoin\test\fuzz\test_runner.py", line 392, in parse_test_list
    test_list_all = subprocess.run(
                    ^^^^^^^^^^^^^^^
  File "C:\hostedtoolcache\windows\Python\3.12.2\x64\Lib\subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command 'D:\a\bitcoin\bitcoin\src\fuzz.exe' returned non-zero exit status 3221226505.
Error: Process completed with exit code 1.
```

(see https://github.com/hebasto/bitcoin/actions/runs/8636782427/job/23677472745)

Do you think this message is meaningful enough to put in to the commit message?

Reply:
On Windows, when `subprocess` sets environment variables, it effectively overrides all of them, which makes `PATH` and `COMSPEC` environment variable empty on the master branch.

Reply:
Linux does not need the `COMSPEC` environment variable to spawn a new process, right?

Reply:
My previous comments were a bit misleading because only [`%SystemRoot%`](https://learn.microsoft.com/en-us/windows/deployment/usmt/usmt-recognized-environment-variables) is required to be passed through into a subprocess.

From the Python [docs](https://docs.python.org/3/library/subprocess.html):
> If specified, `env` must provide any variables required for the program to execute. On Windows, in order to run a [side-by-side assembly](https://en.wikipedia.org/wiki/Side-by-Side_Assembly) the specified `env` must include a valid `SystemRoot`.

Reworked and documented.